shared/src/commonMain/kotlin/com.macaosoftware.sdui.app/plugin/AuthPlugin.kt

```diff
@@ -3,35 +3,113 @@ package com.macaosoftware.sdui.app.plugin
 import com.macaosoftware.plugin.MacaoPlugin
 import com.macaosoftware.sdui.app.util.MacaoError
 import com.macaosoftware.sdui.app.util.MacaoResult
+import dev.gitlive.firebase.Firebase
+import dev.gitlive.firebase.app
+import dev.gitlive.firebase.auth.auth
+import dev.gitlive.firebase.database.database
+import dev.gitlive.firebase.initialize
+import kotlinx.serialization.Serializable
 
 interface AuthPlugin : MacaoPlugin {
     fun initialize()
-    fun signup(signupRequest: SignupRequest)
-    fun login(loginRequest: LoginRequest)
+    suspend fun signup(signupRequest: SignupRequest)
+    suspend fun login(loginRequest: LoginRequest)
 }
 
 /**
  * An empty implementation for those platforms that don't have Firebase.
  * */
 class AuthPluginEmpty : AuthPlugin {
+    private val firebaseAuth = Firebase.auth
```

Comment:
This is supposed to be the implementation in `auth-firebase-gitlive` module, you have to put it there and leave this implementation just for the empty cases.